### PR TITLE
Toolchain: Build fix for macOS

### PR DESCRIPTION
**Pull request by gunnarbeutner**
Unfortunately their sed behaves a tiny bit different to how GNU sed does. The changes from #6394 broke the `BuildIt.sh` script.

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
I've verified that this works on macOS.

File: Toolchain/BuildIt.sh [hunk 1]
```diff
@@ -75,10 +75,16 @@ GCC_NAME="gcc-$GCC_VERSION"
 GCC_PKG="${GCC_NAME}.tar.gz"
 GCC_BASE_URL="http://ftp.gnu.org/gnu/gcc"
 
+SYSTEM_NAME="$(uname -s)"
```
Comment (linusg) [inline Toolchain/BuildIt.sh:78]: this is not used anywhere? Caching the result here is nice as it's not going to change, can you replace the below uname calls with this variable?

Comment (gunnarbeutner) [reply Toolchain/BuildIt.sh:78]: Ah yes, I was planning to do that but then I forgot. Thanks, will do.

Comment (gunnarbeutner) [reply Toolchain/BuildIt.sh]: There we go. Also, I've tested it again on Linux and macOS and it seems to work without problems.

File: Toolchain/BuildIt.sh [hunk 2]
```diff
@@ -78,7 +78,11 @@ GCC_BASE_URL="http://ftp.gnu.org/gnu/gcc"
 buildstep() {
     NAME=$1
     shift
-    "$@" 2>&1 | sed 's|^|\x1b[34m['"${NAME}"']\x1b[39m |'
+    if [ "$(uname -s)" = "Darwin" ]; then
+        "$@" 2>&1 | sed 's|^|['"${NAME}"'] |'
+    else
+        "$@" 2>&1 | sed 's|^|\x1b[34m['"${NAME}"']\x1b[39m |'
+    fi
```
Comment (emanuele6) [inline Toolchain/BuildIt.sh]:
```suggestion
    "$@" 2>&1 | sed $'s|^|\x1b[34m['"${NAME}"$']\x1b[39m |'
```

@gunnarbeutner: I don't have a mac to test this, but wouldn't this work on both mac and linux?

**Pull request merged**